lib/internal/streams/end-of-stream.js

```diff
@@ -199,6 +206,20 @@ function eos(stream, options, callback) {
     });
   }
 
+  if (options.signal && !closed) {
+    const abort = () => callback(new AbortError());
+    if (options.signal.aborted) {
+      process.nextTick(abort);
+    } else {
+      const originalCallback = callback;
+      callback = once((...args) => {
+        options.signal.removeEventListener('abort', abort);
+        ReflectApply(originalCallback, null, args);
+      });
+      options.signal.addEventListener('abort', abort);
+    }
+  }
```

Comment:
please make it call the returning function, so all listeners are removed.

Reply:
Done, also updated the docs.